src/startupscriptgenerator/dotnetcore/scriptgenerator.go

```diff
@@ -53,43 +53,56 @@ isLinuxExecutable() {
 }
 
 cd "$appPath"
-if [ ! -z "$userStartUpCommand" ]; then
-	len=${#userStartUpCommand[@]}
-	if [ "$len" -eq "1" ]; then
+len=${#userStartUpCommand[@]}
+if [ $len -ne 0 ]; then
+	if [ $len -eq 1 ]; then
 		file="${userStartUpCommand[0]}"
 		if [ -f "$file" ]; then
-		  # The startup command could be for example: 'todoApp' or './todoApp'
-		  # So just extract the 'todoApp' part and prefix it with './' to be './todoApp'
-		  startUpCommand="./${file##*/}"
+			startUpCommand="${userStartUpCommand[@]}"
 		else
-		  echo "Could not find the startup file '$file' on disk."
+		  	echo "Could not find the startup file '$file' on disk."
 		fi
-	elif [ "$len" -eq "2" ] && [ "${userStartUpCommand[0]}" == "dotnet" ]; then
+	elif [ $len -eq 2 ] && [ "${userStartUpCommand[0]}" == "dotnet" ]; then
 		if [ -f "${userStartUpCommand[1]}" ]; then
-		  startUpCommand="$userStartUpCommand"
+		  	startUpCommand="${userStartUpCommand[@]}"
+		  	echo "Using the startup command '$startUpCommand' provided by the user to start the app."
 		else
-		  echo "Could not find the file '${userStartUpCommand[1]}' on disk."
+			echo "WARNING: Error in user provided startup command '${userStartUpCommand[@]}'."
+			echo "WARNING: Could not find the file '${userStartUpCommand[1]}' on disk."
 		fi
 	else
-		startUpCommand="$userStartUpCommand"
+		startUpCommand="${userStartUpCommand[@]}"
+		echo "Using the startup command '$startUpCommand' provided by the user to start the app."
 	fi
 fi
```

Comment:
TODO: this should be fixed so that we directly go to using the default app file.